<launch>

  <node name="serial_reader" pkg="my_fused_localization" type="serial_reader" />
  <node name="imu_publisher" pkg="my_fused_localization" type="imu_publisher" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="odom_base_link_broadcaster" args="x y z 0 0 0 odom base_link" />

  <!-- Run the EKF Localization node -->
  <!-- Load the robot_localization node -->
  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" clear_params="true">
    <!-- Configure the input topics for IMU and Odometry -->
    <remap from="imu0" to="/imu_data" />
    <!--remap from="odometry/filtered" to="/noisy_odometry" /-->
    <!-- Set the parameters for the EKF node -->
    <rosparam command="load" file="$(find my_fused_localization)/config/ekf_localization.yaml" />
  </node>
  <node pkg="rosbag" type="record" name="bag_recorder_tf" args="-O output.bag /tf" output="screen" />
  <node pkg="rosbag" type="record" name="bag_recorder_odom" args="-O output.bag /odometry/filtered" output="screen" />
  <node pkg="rosbag" type="record" name="bag_recorder_acc" args="-O output.bag /acceleration/filtered" output="screen" />
  <node pkg="rosbag" type="record" name="bag_recorder_imu" args="-O output.bag /imu_data" output="screen" />

</launch>
